hw/misc/aspeed_hace.c

```diff
@@ -104,6 +104,97 @@ static int do_hash_operation(AspeedHACEState *s, int algo)
     return 0;
 }
 
+static int do_hash_sg_operation(AspeedHACEState *s, int algo)
+{
+    hwaddr src, len, dest, buf_len;
+    struct iovec iov[32];
+    unsigned int i = 0;
+    unsigned int is_last = 0;
+    uint8_t *digest_buf = NULL;
+    size_t digest_len = 0;
+    struct aspeed_sg_list *sg_list;
+    int rc;
+    hwaddr check_len = 0;
+
+    src = s->regs[R_HASH_SRC];
+    len = s->regs[R_HASH_SRC_LEN];
+    dest = s->regs[R_HASH_DEST];
+
+    while (!is_last) {
+        buf_len = sizeof(struct aspeed_sg_list);
+        sg_list = (struct aspeed_sg_list *) address_space_map(&s->dram_as,
```

Comment:
why not use 2 `ldl_le` calls to query the address and the size ?

Reply:
I think I liked the way the address_space_map() was used consistently - first to map the array, and then to map the array contents..

Since the array will always be pairs of u32 values, I guess we could ldl_le instead.. But does that really matters?